### DISCUSSION TITLE
add to fyne (in androidbuild) options how to automatic add permision

### DISCUSSION THREAD

**Discussion by MatejMagat305** [fyne-io/fyne#3972] [Q&A]
I'm working on https://github.com/fyne-io/fyne-x/pull/58/files, it will need permission:
```
<uses-permission android:name="android.permission.BLUETOOTH"/>
<uses-permission android:name="android.permission.BLUETOOTH_ADMIN"/>
<uses-permission android:name="android.permission.ACCESS_FINE_LOCATION"/>
<uses-feature android:name="android.hardware.bluetooth" android:required="false"/>
<uses-feature android:name="android.hardware.bluetooth_le" android:required="false"/>"
```
but it would be waste of source to add to fyne tool to base xml, and little bit annoing make own xml, so maybe similar:  [code](https://github.com/fyne-io/fyne/blob/1834d704a305f4729cce65213f5b5c3098e1d993/cmd/fyne/internal/mobile/build_androidapp.go#L298) - but maybe it would be good to create some principe how to add to xml addition permission, it would working like that all (fyne) packages which would need permision on android would have `android_permission.go` (somethink like fynemetadata) which would contain global variable `permission = {"<uses-permission android:name="android.permission.BLUETOOTH"/>", "...", ...}`

**Comment by andydotxyz** [comment]
I don't think we should put OS specific formatting to fyne build parameters. But a more generic "features" parameter could be helpful so if Bluetooth is turned on we set appropriate permission requests.

**Comment by Bluebugs** [comment]
A possibility would also be to add an OS specific section to the toml file to cover this case.
